addons/bus/static/src/workers/websocket_worker_utils.js

```diff
@@ -40,3 +40,89 @@ export class Deferred extends Promise {
+            try {
+                await logger._ensureDatabaseAvailable();
+                await new Promise((res, rej) => {
+                    const transaction = logger._db.transaction("logs", "readwrite");
+                    const store = transaction.objectStore("logs");
+                    const req = store
+                        .index("timestamp")
+                        .openCursor(IDBKeyRange.upperBound(threshold));
+                    req.onsuccess = (event) => {
+                        const cursor = event.target.result;
+                        if (cursor) {
+                            cursor.delete();
+                            cursor.continue();
+                        }
+                    };
+                    req.onerror = (e) => rej(e.target.error);
+                    transaction.oncomplete = res;
+                    transaction.onerror = (e) => rej(e.target.error);
+                });
+            } catch (error) {
+                console.error(`Failed to clear logs for logger "${logger._name}":`, error);
+            }
+        }
+    }
+
+    constructor(name) {
+        this._name = name;
+        Logger.instances.push(this);
+        Logger.gcOutdatedLogs();
+    }
+
+    async _ensureDatabaseAvailable() {
+        if (this._db) {
+            return;
+        }
+        return new Promise((res, rej) => {
+            const request = indexedDB.open(this._name, 1);
+            request.onsuccess = (event) => {
+                this._db = event.target.result;
+                res();
+            };
+            request.onupgradeneeded = (event) => {
+                if (!event.target.result.objectStoreNames.contains("logs")) {
+                    const store = event.target.result.createObjectStore("logs", {
+                        autoIncrement: true,
+                    });
+                    store.createIndex("timestamp", "timestamp", { unique: false });
+                }
+            };
+            request.onerror = rej;
+        });
+    }
+
+    async log(message) {
+        await this._ensureDatabaseAvailable();
+        const transaction = this._db.transaction("logs", "readwrite");
+        const store = transaction.objectStore("logs");
+        const addRequest = store.add({ timestamp: Date.now(), message });
+        return new Promise((res, rej) => {
+            addRequest.onsuccess = res;
+            addRequest.onerror = rej;
+        });
+    }
+
+    async getLogs() {
+        await Logger.gcOutdatedLogs();
+        await this._ensureDatabaseAvailable();
+        const transaction = this._db.transaction("logs", "readonly");
+        const store = transaction.objectStore("logs");
+        const request = store.getAll();
+        return new Promise((res, rej) => {
+            request.onsuccess = (ev) => res(ev.target.result.map(({ message }) => message));
+            request.onerror = rej;
+        });
+    }
+}
+setInterval(() => Logger.gcOutdatedLogs(), Logger.LOG_TTL);
```

Comment:
Better have it triggered only when there's active logger, rather than all the time.